client/src/components/Map/UCIMap.js

```diff
@@ -45,6 +48,170 @@ export default class UCIMap extends PureComponent {
+                                                        ></Marker>
+                                                    ),
+                                                });
+                                            }}
+                                            onmouseout={function () {
+                                                this.options.map.setState({ info_marker: null });
+                                            }}
+                                        />
+                                    ); // Draw path from last waypoint to next waypoint
+
+                                    poly.push(
+                                        <Polyline
+                                            color={colors[waypointIndex - 1]}
+                                            positions={[
+                                                path[waypointIndex][path[waypointIndex].length - 1],
+                                                coords_array[waypointIndex],
+                                            ]}
+                                            dashArray="4"
+                                        />
+                                    ); // Draw a dashed line directly to waypoint
+                                    let duration =
+                                        obj['routes'][0]['legs'][waypointIndex - 1]['duration'] > 30
+                                            ? Math.round(
+                                                  obj['routes'][0]['legs'][waypointIndex - 1]['duration'] / 60
+                                              ).toString() + ' min'
+                                            : '<1 min';
+                                    let miles =
+                                        (
+                                            Math.floor(
+                                                obj['routes'][0]['legs'][waypointIndex - 1]['distance'] / 1.609 / 10
+                                            ) / 100
+                                        ).toString() + ' mi';
+                                    // Add a marker in the middle (roughly) of the path with miles and walk time
+                                    info_markers.push([
+                                        path[waypointIndex][Math.floor(path[waypointIndex].length / 2)],
+                                        colors[waypointIndex - 1],
+                                        duration,
+                                        miles,
+                                        0.0,
+                                    ]);
+                                }
+                                waypointIndex++;
+                            }
+                        }
+                        this.setState({ poly: poly, info_markers: info_markers, info_marker: null });
+                    });
+                });
+            }
+        } else {
+            this.setState({ poly: [], info_marker: null });
+        }
```

Comment:
Per my previous suggestion, if you always clear the route at the top of the function, you no longer need this else statement.
```suggestion
        }
```